include/wil/safearrays.h

```
@@ -0,0 +1,897 @@
+        template<> struct var_traits<int>                        { static constexpr auto vartype = VT_I4; };
+        template<> struct var_traits<long long>                  { static constexpr auto vartype = VT_I8; };
+        template<> struct var_traits<unsigned char>              { static constexpr auto vartype = VT_UI1; };
+        template<> struct var_traits<unsigned short>             { static constexpr auto vartype = VT_UI2; };
+        template<> struct var_traits<unsigned long>              { static constexpr auto vartype = VT_UI4; };
+        template<> struct var_traits<unsigned int>               { static constexpr auto vartype = VT_UI4; };
+        template<> struct var_traits<unsigned long long>         { static constexpr auto vartype = VT_UI8; };
+        template<> struct var_traits<float>                      { static constexpr auto vartype = VT_R4; };
+        //template<> struct var_traits<double>                     { static constexpr auto vartype = VT_R8; };
+        template<> struct var_traits<VARIANT_BOOL>               { static constexpr auto vartype = VT_BOOL; };
+        template<> struct var_traits<DATE>                       { static constexpr auto vartype = VT_DATE; };
+        template<> struct var_traits<CURRENCY>                   { static constexpr auto vartype = VT_CY; };
+        template<> struct var_traits<DECIMAL>                    { static constexpr auto vartype = VT_DECIMAL; };
+        template<> struct var_traits<BSTR>                       { static constexpr auto vartype = VT_BSTR; };
+        template<> struct var_traits<LPUNKNOWN>                  { static constexpr auto vartype = VT_UNKNOWN; };
+        template<> struct var_traits<LPDISPATCH>                 { static constexpr auto vartype = VT_DISPATCH; };
+        template<> struct var_traits<VARIANT>                    { static constexpr auto vartype = VT_VARIANT; };
+
+        inline void __stdcall SafeArrayDestroy(SAFEARRAY* psa) WI_NOEXCEPT

```

Specifying the calling convention seems unnecessary, or am I missing something?

---

See [details::SetEvent](https://github.com/microsoft/wil/blob/3c00e7f1d8cf9930bbb8e5be3ef0df65c84e8928/include/wil/resource.h#L2328) in `resource.h`.  That's what I modeled it after.  It specifies the calling convention, presumably to make an identical (except for fail fast) version of the API.

---

I suspect that was just because of copy/paste, but with `extern "C"` functions, you never know... Do you get compiler/linker errors with it removed?